MapboxCoreNavigation/RouteController.swift

```diff
@@ -628,6 +628,10 @@ extension RouteController: CLLocationManagerDelegate {
         let currentStepProgress = routeProgress.currentLegProgress.currentStepProgress
         let currentStep = currentStepProgress.step
 
+        let intersectionDistances = routeProgress.currentLegProgress.currentStepProgress.intersectionDistances
```

Comment:
If the user has set `NavigationViewController.automaticallyChangeStyleForTunnel` to false, is it necessary to perform these calculations?

Reply:
@bsudekum FYI 
- [ ] It shouldn't be restricted to automaticallyChangeStyleForTunnel. It should be wrapped within a isDeadReckoningEnabled flag.
- [ ] This calculation updates the `intersectionIndex` - fix for issue https://github.com/mapbox/mapbox-navigation-ios/issues/1170. @1ec5 do you need to use this fix for a task unrelated to dead reckoning?

Reply:
@vincethecoder I don't think this code should be intertwined with `isDeadReckoningEnabled`. Let's keep these as separate ideas for now.

IMO, the fix here is to move `NavigationViewController.automaticallyChangeStyleForTunnel ` to `RouteController` and rename it to something more appropriate.

Reply:
@bsudekum I believe it was mentioned in one of my PR comment to separate the UI-related things in `NavigationViewController` from `RouteController`. Now, I'm totally thrown for loop 🤔  
cc @1ec5 - do you recall this discussion?

Reply:
Yes, in https://github.com/mapbox/mapbox-navigation-ios/pull/1127#discussion_r172624263 I recommended keeping UI-related stuff out of Core Navigation. I stand by that recommendation.